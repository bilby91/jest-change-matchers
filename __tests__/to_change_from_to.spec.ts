import "jest"
import "../"

let sharedState = 0

describe("toChangeFromTo", () => {
  beforeEach(() => {
    sharedState = 0
  })

  describe(".toChangeFromTo", () => {
    it("passes when it changes from previous and new value", async () => {
      await expect(async () => {
        sharedState += 1
      }).toChangeFromTo(async () => {
        return sharedState
      }, 0, 1)
    })

    it("throws an error when it doesn't change from previous value", async () => {
      const testCase = () => {
        return expect(async () => {
          sharedState += 2
        }).toChangeFromTo(async () => {
          return sharedState
        }, 1, 2)
      }

      await expect(testCase()).rejects.toThrowErrorMatchingSnapshot()
    })

    it("throws an error when it doesn't change to new value", async () => {
      const testCase = () => {
        return expect(async () => {
          sharedState += 2
        }).toChangeFromTo(async () => {
          return sharedState
        }, 0, 1)
      }

      await expect(testCase()).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe(".not.toChangeFromTo", () => {
    it("passes when it doesn't change to previous and new value", async () => {
      await expect(async () => {
        sharedState += 1
      }).not.toChangeFromTo(async () => {
        return sharedState
      }, 0, 0)
    })

    it("throws an error when it change to previous and new value", async () => {
      const testCase = () => {
        return expect(async () => {
          sharedState += 2
        }).not.toChangeFromTo(async () => {
          return sharedState
        }, 0, 2)
      }

      await expect(testCase()).rejects.toThrowErrorMatchingSnapshot()
    })

  })
})
