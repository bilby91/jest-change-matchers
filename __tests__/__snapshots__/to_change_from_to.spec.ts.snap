// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toChangeFromTo .not.toChangeFromTo throws an error when it change to previous and new value 1`] = `
"Expected target not to have changed
  from: 0
  to: 2
Received:
  from: 0
  to: 2
"
`;

exports[`toChangeFromTo .toChangeFromTo throws an error when it doesn't change from previous value 1`] = `
"Expected target to have changed
  from: 1
  to: 2
Received:
  from: 0
  to: 2
"
`;

exports[`toChangeFromTo .toChangeFromTo throws an error when it doesn't change to new value 1`] = `
"Expected target to have changed
  from: 0
  to: 1
Received:
  from: 0
  to: 2
"
`;
